src/env-inl.h

```diff
@@ -101,6 +101,20 @@ inline void Environment::AsyncHooks::set_enable_callbacks(uint32_t flag) {
   fields_[kEnableCallbacks] = flag;
 }
 
+inline Environment::AsyncCallbackScope::AsyncCallbackScope(Environment* env)
+    : env_(env) {
+  env_->async_callback_cntr_++;
+}
+
+inline Environment::AsyncCallbackScope::~AsyncCallbackScope() {
+  env_->async_callback_cntr_--;
+  CHECK_GE(env_->async_callback_cntr_, 0);
+}
+
+inline bool Environment::AsyncCallbackScope::in_tick() {
```

Comment:
@trevnorris  Is `in_tick` a name that actually describes what `env_->async_callback_cntr_ > 1` means? I thought that `env_->async_callback_cntr_ > 1` meant that there's currently nested "asynchronous callbacks" that are active, so I would think that a name such as `nested_async_callback_active` would be more appropriate, but I may be missing something.

I also failed to see where `*::MakeCallback` is called recursively with the changes in https://github.com/nodejs/node/pull/4509, but that's due to my lack of knowledge of the HTTP parser. Would you mind giving me an example of code that triggers a recursive call to `*::MakeCallback`?